actions/assistant.py

```diff
@@ -169,13 +169,20 @@ def _install_repo(repo: Dict[str, str], remove_dir: bool = True) -> List[str]:
         if "checkout" in repo:
             assert isinstance(repo["checkout"], str)
             cmds.append(f"git checkout {repo['checkout']}")
-        pip_install = "."
-        if "install_extras" in repo:
-            pip_install += f"[{AssistantCLI._extras(repo['install_extras'])}]"
-        cmds.append(f"pip install --quiet {pip_install}")
-        if "install_file" in repo:
-            assert isinstance(repo["install_file"], str)
-            cmds.append(f"pip install --quiet --upgrade -r {repo['install_file']}")
+
+        cmds += AssistantCLI.before_commands(repo, stage="install", as_append=True)
```

Comment:
Seems like the change in assistant.py throws an error:
```
Traceback (most recent call last):
  File "actions/assistant.py", line 285, in <module>
    fire.Fire(AssistantCLI)
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "actions/assistant.py", line 234, in prepare_env
    script += AssistantCLI._install_repo(repo, remove_dir=False)
  File "actions/assistant.py", line 173, in _install_repo
    cmds += AssistantCLI.before_commands(repo, stage="install", as_append=True)
  File "actions/assistant.py", line 204, in before_commands
    config = AssistantCLI._load_config(config_file)
  File "actions/assistant.py", line 78, in _load_config
    assert os.path.isfile(config_file), f"Missing config file: {config_file}"
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/genericpath.py", line 30, in isfile
    st = os.stat(path)
TypeError: stat: path should be string, bytes, os.PathLike or integer, not dict
```
https://github.com/PyTorchLightning/ecosystem-ci/runs/5059997629?check_suite_focus=true

Reply:
yeah, want to merge #36 and then continue with this one :rabbit: